<launch>

    <!-- Params -->
    <arg name="rate" default="192000"/>
    <arg name="channels" default="2"/>
    <arg name="pcm_id" default="hw:1,0"/>
    <arg name="pcm_frames" default="100"/>
    <arg name="pcm_bits" default="32"/>
    <arg name="msg_length" default="1.0"/>

    <arg name="path" default="/root/SSD/"/> <!-- cmdline params -->

    <!-- Focusrite -->
    <node name="focusrite" pkg="recorder" type="focusrite" output="screen" clear_params="true">
        <param name="pcm_id" type="str" value="$(arg pcm_id)"/>
        <param name="pcm_frames" type="int" value="$(arg pcm_frames)"/>
        <param name="pcm_sampleRate" type="int" value="$(arg rate)"/>
        <param name="pcm_available_channels" type="int" value="$(arg channels)"/>
        <param name="pcm_bits" type="int" value="$(arg pcm_bits)"/>
        <param name="msg_length" type="double" value="$(arg msg_length)"/>
    </node>

    <!-- wav -->
    <node pkg="recorder" type="wav.py" name="wav_node" output="screen">
        <param name="channels" type="int" value="$(arg channels)"/>
        <param name="rate" type="int" value="$(arg rate)"/>
        <param name="path" type="str" value="$(arg path)"/>
    </node>



</launch>
